### PR TITLE
[DDS-1576] Updated ci runner

### PR DESCRIPTION
**Pull request by GROwen**
https://digital-vic.atlassian.net/browse/DDS-1593
This PR sets up an image build pipeline in GitHub Actions workflows that has feature parity with the workflows in CircleCI.

* Moves the image build process to GitHub Actions
The workflows [goss-tests](https://github.com/dpc-sdp/bay/blob/build/DDS-1576--updated-ci-runner/.github/workflows/goss-tests.yml) and [vulnerability-scan-build](https://github.com/dpc-sdp/bay/blob/build/DDS-1576--updated-ci-runner/.github/workflows/vulnerability-scan-build.yml) will only run after build-deploy has [completed](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onworkflow_runbranchesbranches-ignore). This will also **only** occur once the workflows are in the default branch.
* Migrates vulnerability scanning from Clair to Trivy
* Adds dynamic labels
```
"Labels": {
    "maintainer": "Digital Victoria",
    "org.opencontainers.image.authors": "Digital Victoria",
    "org.opencontainers.image.created": "2023-08-01T03:17:56.179Z",
    "org.opencontainers.image.description": "PHP-FPM image for Bay container platform",
    "org.opencontainers.image.licenses": "",
    "org.opencontainers.image.revision": "97f7e1f4b843c34aaa93656b8aeb50d80404293d",
    "org.opencontainers.image.source": "https://github.com/dpc-sdp/bay/blob/5.x/images/bay-php/Dockerfile.fpm",
    "org.opencontainers.image.title": "php-fpm",
    "org.opencontainers.image.url": "https://github.com/dpc-sdp/bay",
    "org.opencontainers.image.version": "build-DDS-1576--updated-ci-runner",
    "repository": "git://github.com/dpc-sdp/bay.git"
},
```
A caveat to this, according to [GitHub's image tag detail page multi-arch images](https://github.com/dpc-sdp/bay/pkgs/container/bay%2Fphp-fpm/114500372?tag=build-DDS-1576--updated-ci-runner) require the description to be set using an annotation on the manifest rather than a label. This isn't supported by our build process (buildx bake-action) at the moment but there is a [proposal to make it possible](https://github.com/docker/buildx/issues/1171).
* [Goss tests are passing](https://github.com/dpc-sdp/bay/actions/runs/5710095404)

### TODO
* Once this PR is merged the repository settings will need to be updated to remove the CircleCI check
* Follow up annotations for setting multi-arch descriptions

### CODE REVIEW

File: .github/workflows/build-deploy.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+name: build-deploy-bay-images
+run-name: Build and deploy Bay images
+on:
+  schedule:
+    - cron: '23 20 * * 0'
```
Comment (GROwen) [inline .github/workflows/build-deploy.yml:5]: Schedules only run on the default branch. `5.x` in this instance.

File: .github/workflows/build-deploy.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  workflow_dispatch:
```
Comment (GROwen) [inline .github/workflows/build-deploy.yml:6]: This event allows for workflows to be triggered [manually via the GitHub UI](https://github.blog/changelog/2020-07-06-github-actions-manual-triggers-with-workflow_dispatch/). The UI feature is only visible if the workflow file is on the default branch, so we won't see it until this PR is merged.

File: .github/workflows/build-deploy.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      matrix:
+        images: ${{ fromJson(vars.IMAGES) }}
+    steps:
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - uses: actions/checkout@v3
```
Comment (GROwen) [inline .github/workflows/build-deploy.yml:26]: Necessary to make the bake file available to a subsequent action.

**Review by nicksantamaria — CHANGES_REQUESTED** [review]
@GROwen Can we get a new section in README with steps on adding new images? I spent a bit of time figuring out how the matrix works, and it depends on an [actions variable](https://github.com/dpc-sdp/bay/settings/variables/actions/IMAGES) which isn't obvious from the outset.

**Review by nicksantamaria — APPROVED** [review]

**Pull request merged**